threefoldtech/tfgrid-sdk-ts
# Fix: CapRover Dashboard App: Details of Workers disappear after few seconds

### Description

Fix listing Caprover workers

### Changes

- Send both of the deployments, leader, and workers.
- Call the 'getGrafanaUrl' on mount the component.
- Delete the full cluster.
- Ensure that workers are correctly removed from the deployment list when the leader is deleted to maintain data integrity.
- Merged the latest changes from the development branch to keep the codebase up to date.
- Conducted thorough testing to verify the complete functionality and ensure no regressions were introduced.

### Test Scenarios

1. Deploy `leader`, and `worker`, Go go to the contracts page and delete the deployed `leader`.
2. Deploy `leader`, attach a `worker` to it then delete the whole cluster just in time without refreshing the page.
3. Deploy `leader`, and `worker`, delete the deployed `worker` and attach another, then delete the whole cluster just in time without refreshing the page.
4. Deploy a cluster with 1 leader and 2 workers check the deployment listing dialog and make sure all of them are listed
3. Delete a cluster with the leader and more than 3 workers and make sure all of them are deleted.

PS: Make sure no errors on the Caprover listing page.

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3281

### Screenshots/Videos

[Screencast from 09-03-2024 05:27:14 PM.webm](https://github.com/user-attachments/assets/45752f9d-d688-4869-8327-f68c5aad1c65)

![image](https://github.com/user-attachments/assets/51593928-7848-4744-b01d-f7d4474a3c57)

### Checklist
